src/libraries/Native/Unix/System.Globalization.Native/pal_icushim.c

```diff
@@ -42,6 +42,53 @@ static int FindICULibs()
     return TRUE;
 }
 
+#elif defined(TARGET_ANDROID)
+
+// support ICU versions from 0-255
+#define MinICUVersion 50
+#define MaxICUVersion 255
+#define MaxICUVersionStringLength 4
+
+static int FindSymbolVersion(char* symbolName, char* symbolVersion)
+{
+    for (int i = MinICUVersion; i <= MaxICUVersion; i++)
+    {
+        sprintf(symbolVersion, "_%d", i);
+        sprintf(symbolName, "u_strlen%s", symbolVersion);
+        if (dlsym(libicuuc, symbolName) != NULL)
+        {
+            return TRUE;
+        }
+    }
+
+    return FALSE;
+}
+
+static int FindICULibs(char* symbolName, char* symbolVersion)
```

Comment:
/cc @safern @tarekgh FYI regarding our discussions about ICU, this is one variant that loads the system ICU.

Reply:
This conflicts with what I'm doing here: https://github.com/dotnet/runtime/pull/33704/files#diff-9812d22f4140085b5587ec72c0d99cb0R29

Can you hold your PR until I merge mine? It is close to be merged (I hope).

Reply:
Sure

Reply:
CC @janvorli

Reply:
@akoeplinger so we can assume that there is only one ICU version installed on Android, right? The reason why we scan in the opposite direction on Linux is to get the highest version installed, so I just want to make sure we don't need to care about it on Android.

Reply:
@janvorli yes